gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -230,15 +230,16 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
       Map<String, String> flowMetadata = TimingEventUtils.getFlowMetadata(flowSpec);
       String flowExecutionId = String.valueOf(FlowUtils.getOrCreateFlowExecutionId(flowSpec));
 
+      // Note we use "" as jobName for lease arbitration since orchestration is at a flow level
       DagActionStore.DagAction flowAction =
-          new DagActionStore.DagAction(flowGroup, flowName, flowExecutionId, DagActionStore.FlowActionType.LAUNCH);
+          new DagActionStore.DagAction(flowGroup, flowName, flowExecutionId, "", DagActionStore.DagActionType.LAUNCH);
 
       // If multi-active scheduler is enabled do not pass onto DagManager, otherwise scheduler forwards it directly
       // Skip flow compilation as well, since we recompile after receiving event from DagActionStoreChangeMonitor later
-      if (flowTriggerHandler.isPresent()) {
+      if (flowTriggerDecorator.isPresent()) {
 
         // Adopt consensus flowExecutionId for scheduled flows
-        flowTriggerHandler.get().handleTriggerEvent(jobProps, flowAction, triggerTimestampMillis, isReminderEvent,
+        flowTriggerDecorator.get().handleTriggerEvent(jobProps, flowAction, triggerTimestampMillis, isReminderEvent,
```

Comment:
naming-wise it doesn't really seem like it's "handling trigger events".  if anything that's what the calling `Orchestrator::orchestrate` is doing.  what about `handleFlowLaunchTrigger`?

(post-reminders, we can't any longer treat "trigger" as a synonym for "flow launch".)